src/parser/mod.rs

```diff
@@ -4294,7 +4312,9 @@ impl<'a> Parser<'a> {
         let mut data_type = self.parse_data_type()?;
         if let DataType::Custom(n, _) = &data_type {
             // the first token is actually a name
-            name = Some(n.0[0].clone());
+            match n.0[0].clone() {
+                ObjectNamePart::Identifier(ident) => name = Some(ident),
```

Comment:
Once we start adding more to the `ObjectNamePart` enum, we will return parsing error for the other variants here.